Sources/SwiftFusion/Inference/ValuesStorage.swift

```diff
@@ -35,6 +35,13 @@ class AnyDifferentiableStorage: AnyArrayStorage {
   }
 }
 
+extension AnyArrayBuffer where Storage: AnyDifferentiableStorage {
+  mutating func move(along directionsStart: UnsafeRawPointer) {
+    ensureUniqueStorage()
```

Comment:
As noted in the other PR, I'm skeptical of this approach.  Why initialize the storage and then update it?

I note that “copy on write” can be a somewhat misleading phrase ;-)